### DISCUSSION TITLE
How to make a mask map?

### DISCUSSION THREAD

**Discussion by Abicula** [iiasa/CWatM#71] [Q&A] [answered]
I am trying to run the model for a an extended region, and I have been stuck on trying to have a maskmap.
My region is Saudi Arabia, and I have been trying to recreate a tif file similar to the "rhine30min." 
Can you provide me with some guidance?

I have tried ArcGIS Pro, and I have tried to find an online georeferenced map for Saudi, but the data for that is scarce and it has not been working.

I have tried multiple ways of creating the TIFF file (such as downloading shape file with Saudi borders and rasterize it, then export it as TIFF) I believe this was my best attempt, and I keep getting "Error 101"

If anyone has any idea on how to create a maskmap, please send help..

**Comment by mikhailsmilovic** [comment, marked as answer]
Guten Tag, this video (https://youtu.be/AH4RA0G4rG4) now goes through 
creating a mask map by
rasterizing and aligning a shapefile
based on the resolution and alignment of another CWatM input raster file
using ArcMap.

**Reply by Abicula** [reply]
Hi Mikhail,

Thank you for creating the youtube video, it was really helpful and easy to follow. 
I was able to create georeferenced tif file (you can find it in the end of this reply), but I still cannot use it as a domain, and I am getting an error message. My goal for now is to create a discharge *map,* (something similar to this) but only for SA, and then I will downscale:

![image](https://github.com/iiasa/CWatM/assets/123258196/02898557-80bd-4854-81d4-221ba2b9e994)



When I use only number as coordinates to create the domain and gauges, the simulation works (maskmap & gauge as coordinates)):

![image](https://github.com/iiasa/CWatM/assets/123258196/11349b6e-9144-4220-aa62-fc9430a236b2)


However, when I try to input my tif map as a domain, it doesn't work (maskmap as tif file, gauge as coordinate):

![image](https://github.com/iiasa/CWatM/assets/123258196/93f0c5c3-69dd-4be0-b27b-47443bda2019)


(maskmap & gauge as tif file):

![image](https://github.com/iiasa/CWatM/assets/123258196/b1c714ce-ea6a-49c9-93b1-934dd21c520b)

And I get the following Error Message:

![image](https://github.com/iiasa/CWatM/assets/123258196/5ddbad01-be1a-481e-9167-ea52d43e4ba7)

If I understand correctly, my .tif file is cells that are georeferenced (so it should have coordintes) and CWatM should be able to read the coordinates of each cell and calculate discharge. Please correct me if my understanding is wrong.


Tif file used as a domain & gauge that is giving the error message:
https://drive.google.com/file/d/1iWjjQ3TF7jmellebxeCED1ksCWjv8CNB/view?usp=sharing

P.S., I only have ArcGIS Pro, which was used to produce the mask. I noticed a couple of differences between my produced map and rhine30min map

**Reply by mikhailsmilovic** [reply]
Guten Morgen, Gauges holds the cell locations for time series text format (TSS) output. 

**If we want output at all cells**, the simplest option is to make Gauges equal to the location of _any_ coordinate within your model domain. Then, include the variable of interest within the map outputs for the timestep of interest, for example,
Gauges = _longitude1 latitude1_
OUT_MAP_Daily = _output_variable_

**If we want a text time series for specific locations**, Gauges should hold a sequence of longitudes and latitudes or point to a .tif where cells of interest are numbered 1, 2, 3, ...
Gauges =  _longitude1 latitude1 longitude 2 latitude 2 longitude 3 latitude 3_
OR, Gauges = _numbered_list_of_locations.tif_
Then, OUT_TSS_Daily = _output_variable_
